src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -190,6 +206,28 @@ internal Invocation CreateInvocation(InvocationVersionOne v1Invocation)
             return invocation;
         }
 
+        internal Location CreateLocation(LocationVersionOne v1Location)
+        {
+            Location location = null;
+
+            if (v1Location != null)
+            {
+                location = new Location
+                {
+                    FullyQualifiedLogicalName = v1Location.LogicalLocationKey ?? v1Location.FullyQualifiedLogicalName,
+                    PhysicalLocation = CreatePhysicalLocation(v1Location.ResultFile),
+                    Properties = v1Location.Properties
+                };
+
+                if (_currentRun.LogicalLocations.ContainsKey(location.FullyQualifiedLogicalName))
+                {
+                    _currentRun.LogicalLocations[location.FullyQualifiedLogicalName].DecoratedName = v1Location.DecoratedName;
+                }
```

Comment:
If the entry doesn't exist, we should create it, right?

Reply:
The logicalLocations dictionary has been imported at this point, wouldn't that definitely include it?

Reply:
Not necessarily.  `run.logicalLocations` is optional (in both v1 and v2). Now, there's nowhere else to put `decoratedName`, so if you have `decoratedName`, you have to make sure `run.logicalLocations` is present. Come to think of it, you need to check for the existence of the dictionary before adding to it. And I suspect you'll be doing that check-for-dictionary-and-create-if-necessary-and-then-add-entry multiple times, so you might want to factor it out.

Reply:
Yeah I already have AddLogicalLocation which should work here.

Reply:
Done